pkg/controller/reconciler/annotations_test.go

```diff
@@ -0,0 +1,148 @@
+		want       deploymentAnnotations
+		wantErr    bool
+	}{
+		{
+			name: "radius-disabled-with-annotation",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						AnnotationRadiusEnabled: "false",
+					},
+				},
+			},
+			want: deploymentAnnotations{
+				Configuration:     nil,
+				ConfigurationHash: "",
+				Status:            nil,
+			},
+			wantErr: false,
+		},
+		{
+			name: "radius-disabled-empty-annotation-map",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{},
+				},
+			},
+			want: deploymentAnnotations{
+				Configuration:     nil,
+				ConfigurationHash: "",
+				Status:            nil,
+			},
+			wantErr: false,
+		},
+		{
+			name: "radius-disabled-no-annotations",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{},
+			},
+			want: deploymentAnnotations{
+				Configuration:     nil,
+				ConfigurationHash: "",
+				Status:            nil,
+			},
+			wantErr: false,
+		},
+		{
+			name: "radius-was-enabled-now-disabled-with-annotations",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						AnnotationRadiusEnabled:           "false",
+						AnnotationRadiusConfigurationHash: "configuration-hash",
+						AnnotationRadiusStatus:            string(dsm),
+					},
+				},
+			},
+			want: deploymentAnnotations{
+				Configuration:     nil,
+				ConfigurationHash: "configuration-hash",
+				Status:            ds,
+			},
+			wantErr: false,
+		},
+		{
+			name: "radius-enabled-with-annotations",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						AnnotationRadiusEnabled:                              "true",
+						AnnotationRadiusConfigurationHash:                    "configuration-hash",
+						AnnotationRadiusStatus:                               string(dsm),
+						AnnotationRadiusApplication:                          "test-application",
+						AnnotationRadiusEnvironment:                          "test-environment",
+						AnnotationRadiusConnectionPrefix + "test-connection": "test-connection-value",
+					},
+				},
+			},
+			want: deploymentAnnotations{
+				Configuration: &deploymentConfiguration{
+					Environment: "test-environment",
+					Application: "test-application",
+					Connections: map[string]string{
+						"test-connection": "test-connection-value",
+					},
+				},
+				ConfigurationHash: "configuration-hash",
+				Status:            ds,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := readAnnotations(tt.deployment)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("readAnnotations() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("readAnnotations() = %v, want %v", got, tt.want)
```

Comment:
Please don't write this style of test comparison. See guidance here: https://github.com/radius-project/radius/tree/main/docs/contributing/contributing-code/contributing-code-writing#appendix-style-guidance-we-dont-follow